lib/dns/dns.go

```diff
@@ -639,3 +640,94 @@ func (s *Server) Stop() {
 func (s *Server) Wait() {
 	s.wg.Wait()
 }
+
+type FQDNs map[string]bool
+
+// ReverseLookup returns a set of FQDNs for ipAddr from nameservers in /etc/resolv.conf
+// /etc/hosts and /etc/nsswitch.conf are ignored by this function
+func ReverseLookup(ipAddr string) (domains FQDNs) {
+	domains = make(map[string]bool)
+
+	address, err := reverseaddr(ipAddr)
+	if err != nil {
+		log.Errorf("%s", err)
+		return
+	}
+
+	nameservers := resolvconf()
+	log.Infof("Looking up rDNS record for %s\n", address)
+	for _, n := range nameservers {
+		dnsClient := new(mdns.Client)
+		msg := new(mdns.Msg)
+
+		msg.SetQuestion(address, mdns.TypePTR)
+		r, _, err := dnsClient.Exchange(msg, fmt.Sprintf("%s:53", n))
```

Comment:
btw, for the cases like this `n + ":53"` is enough and it is even faster. if you need more elements, then your approach is better.

Reply:
ORLY? I did not know that! Thanks I'll change it to the simpler `+` notation.